csrc/multidevice/lower_communication.cpp

```diff
@@ -295,43 +204,24 @@ CommParams createParamsForReduce(
     DeviceIdxType root,
     TensorView* input_tv,
     TensorView* output_tv,
-    at::Tensor input_tensor,
-    at::Tensor output_tensor,
     BinaryOpType op_type) {
   const DeviceMesh& mesh = input_tv->getDeviceMesh();
   CommParams params;
   params.root = root;
   params.redOp = getC10dReduceOpType(op_type);
   params.team = mesh.vector();
-  bool is_root_in_mesh = mesh.has(root);
-  if (!is_root_in_mesh) {
+  if (!mesh.has(root)) {
+    params.is_root_in_mesh = false;
     params.team.push_back(root);
   }
-
-  auto sharded_dim = output_tv->getReductionAxis().value();
-  if (mesh.has(my_device_index)) {
-    params.src_bufs = {input_tensor.squeeze(sharded_dim)};
-  }
-
-  if (my_device_index == root) {
-    params.dst_bufs = {output_tensor};
-    // The reduce semantics imposes the root to be both
-    // sender and receiver. If the root is not in the mesh, we thus
-    // have to artificially make it send and receive a dummy buffer
-    if (!is_root_in_mesh) {
-      at::Tensor dummy = createDummyTensor(output_tensor, op_type);
-      params.src_bufs.push_back(dummy);
-    }
-  }
+  // FIXME: we may want to store sharded_dim to params for speed.
```

Comment:
I am not sure to understand this FIXME, can you please explain?
In general, the `Communication` class should store pointers to I/O TensorView, from which all the sharding info can be infered

Reply:
My bad -- this FIXME is outdated and should be removed. 

> In general, the Communication class should store pointers to I/O TensorView, from which all the sharding info can be infered

TensorView is available at compileFusion time, so there's a benefit of precompute things over there. However, I haven't timed this so I'm happy to revisit the decision later.